// parsen/lib/parse/objects.js
// Generated by CoffeeScript 1.4.0
var createObject, deleteObject, getObject, queryObjects, request, serialize, updateObject,
  _this = this;

request = require('request');

serialize = require('../utils/utils');

createObject = function(className, data, callback) {
  if (data == null) {
    data = {};
  }
  if (!_this.ready) {
    throw new Error(_this.noKeys);
  }
  return request({
    method: 'POST',
    uri: "https://api.parse.com/1/classes/" + className,
    headers: {
      "X-Parse-Application-Id": _this.parseApplicationId,
      "X-Parse-REST-API-Key": _this.parseRESTAPIKey,
      "Content-Type": "application/json"
    },
    body: JSON.stringify(data)
  }, function(error, response, body) {
    return callback(error, response, body);
  });
};

getObject = function(className, objectId, callback) {
  if (!_this.ready) {
    throw new Error(_this.noKeys);
  }
  return request({
    method: 'GET',
    uri: "https://api.parse.com/1/classes/" + className + "/" + objectId,
    headers: {
      "X-Parse-Application-Id": _this.parseApplicationId,
      "X-Parse-REST-API-Key": _this.parseRESTAPIKey,
      "Content-Type": "application/json"
    }
  }, function(error, response, body) {
    return callback(error, response, body);
  });
};

updateObject = function(className, objectId, data, callback) {
  if (!_this.ready) {
    throw new Error(_this.noKeys);
  }
  return request({
    method: 'PUT',
    uri: "https://api.parse.com/1/classes/" + className + "/" + objectId,
    headers: {
      "X-Parse-Application-Id": _this.parseApplicationId,
      "X-Parse-REST-API-Key": _this.parseRESTAPIKey,
      "Content-Type": "application/json"
    },
    body: JSON.stringify(data)
  }, function(error, response, body) {
    return callback(error, response, body);
  });
};

deleteObject = function(className, objectId, callback) {
  if (!_this.ready) {
    throw new Error(_this.noKeys);
  }
  return request({
    method: 'DELETE',
    uri: "https://api.parse.com/1/classes/" + className + "/" + objectId,
    headers: {
      "X-Parse-Application-Id": _this.parseApplicationId,
      "X-Parse-REST-API-Key": _this.parseRESTAPIKey,
      "Content-Type": "application/json"
    }
  }, function(error, response, body) {
    return callback(error, response, body);
  });
};

queryObjects = function(className, params, callback) {
  if (!_this.ready) {
    throw new Error(_this.noKeys);
  }
  return request({
    method: 'GET',
    uri: "https://api.parse.com/1/classes/" + className + "?" + (serialize(params)),
    headers: {
      "X-Parse-Application-Id": _this.parseApplicationId,
      "X-Parse-REST-API-Key": _this.parseRESTAPIKey,
      "Content-Type": "application/json"
    }
  }, function(error, response, body) {
    return callback(error, response, body);
  });
};

module.exports.createObject = createObject;

module.exports.getObject = getObject;

module.exports.updateObject = updateObject;

module.exports.deleteObject = deleteObject;

module.exports.queryObjects = queryObjects;
