// parsen/lib/parse/users.js
// Generated by CoffeeScript 1.4.0
var createUser, deleteUser, getUser, loginUser, queryUsers, request, serialize, updateUser,
  _this = this;

request = require('request');

serialize = require('../utils/utils');

createUser = function(username, password, data, callback) {
  if (data == null) {
    data = {};
  }
  if (!_this.ready) {
    throw new Error(_this.noKeys);
  }
  data.username = username;
  data.password = password;
  return request({
    method: 'POST',
    uri: "https://api.parse.com/1/users",
    headers: {
      "X-Parse-Application-Id": _this.parseApplicationId,
      "X-Parse-REST-API-Key": _this.parseRESTAPIKey,
      "Content-Type": "application/json"
    },
    body: JSON.stringify(data)
  }, function(error, response, body) {
    return callback(error, response, body);
  });
};

loginUser = function(username, password, callback) {
  var params;
  if (!_this.ready) {
    throw new Error(_this.noKeys);
  }
  params = {
    "username": username,
    "password": password
  };
  return request({
    method: 'GET',
    uri: "https://api.parse.com/1/login?" + (serialize(params)),
    headers: {
      "X-Parse-Application-Id": _this.parseApplicationId,
      "X-Parse-REST-API-Key": _this.parseRESTAPIKey,
      "Content-Type": "application/json"
    }
  }, function(error, response, body) {
    return callback(error, response, body);
  });
};

getUser = function(userId, callback) {
  if (!_this.ready) {
    throw new Error(_this.noKeys);
  }
  return request({
    method: 'GET',
    uri: "https://api.parse.com/1/users/" + userId,
    headers: {
      "X-Parse-Application-Id": _this.parseApplicationId,
      "X-Parse-REST-API-Key": _this.parseRESTAPIKey,
      "Content-Type": "application/json"
    }
  }, function(error, response, body) {
    return callback(error, response, body);
  });
};

updateUser = function(userId, sessionToken, data, callback) {
  if (!_this.ready) {
    throw new Error(_this.noKeys);
  }
  return request({
    method: 'PUT',
    uri: "https://api.parse.com/1/users/" + userId,
    headers: {
      "X-Parse-Application-Id": _this.parseApplicationId,
      "X-Parse-REST-API-Key": _this.parseRESTAPIKey,
      "X-Parse-Session-Token": sessionToken,
      "Content-Type": "application/json"
    },
    body: JSON.stringify(data)
  }, function(error, response, body) {
    return callback(error, response, body);
  });
};

deleteUser = function(userId, sessionToken, callback) {
  if (!_this.ready) {
    throw new Error(_this.noKeys);
  }
  return request({
    method: 'DELETE',
    uri: "https://api.parse.com/1/users/" + userId,
    headers: {
      "X-Parse-Application-Id": _this.parseApplicationId,
      "X-Parse-REST-API-Key": _this.parseRESTAPIKey,
      "X-Parse-Session-Token": sessionToken,
      "Content-Type": "application/json"
    }
  }, function(error, response, body) {
    return callback(error, response, body);
  });
};

queryUsers = function(params, callback) {
  if (!_this.ready) {
    throw new Error(_this.noKeys);
  }
  return request({
    method: 'GET',
    uri: "https://api.parse.com/1/users?" + (serialize(params)),
    headers: {
      "X-Parse-Application-Id": _this.parseApplicationId,
      "X-Parse-REST-API-Key": _this.parseRESTAPIKey,
      "Content-Type": "application/json"
    }
  }, function(error, response, body) {
    return callback(error, response, body);
  });
};

module.exports.createUser = createUser;

module.exports.loginUser = loginUser;

module.exports.getUser = getUser;

module.exports.updateUser = updateUser;

module.exports.deleteUser = deleteUser;

module.exports.queryUsers = queryUsers;
